nav2_behavior_tree/include/nav2_behavior_tree/json_utils.hpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__JSON_UTILS_HPP_
+#define NAV2_BEHAVIOR_TREE__JSON_UTILS_HPP_
+
+#include <string>
+#include <set>
+#include <vector>
+
+#include "rclcpp/time.hpp"
+#include "rclcpp/node.hpp"
+#include "behaviortree_cpp/json_export.h"
+#include "geometry_msgs/msg/point.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "geometry_msgs/msg/quaternion.hpp"
+#include "nav_msgs/msg/goals.hpp"
+#include "nav_msgs/msg/path.hpp"
+
+// The follow templates are required when using Groot 2 to visualize the BT. They
+// convert the data types into JSON format easy for visualization.
+
+namespace builtin_interfaces::msg
+{
+
```

Comment:
Add in https://github.com/ros-navigation/navigation2/blob/main/nav2_msgs/msg/MissedWaypoint.msg too (PR https://github.com/ros-navigation/navigation2/pull/4994 to use in a modified form shortly)